lambda/helper/python/comprehendHelper.py

```diff
@@ -0,0 +1,454 @@
+import boto3
+from botocore.exceptions import ClientError
+import json
+import threading
+from helper import S3Helper
+
+
+class ComprehendHelper:
+
+    def getNumOfPages(self,
+                      textract):
+        return textract[0]['DocumentMetadata']['Pages']
+
+    #
+    # this function iterates and extracts all textual elements, aka blocks, that were
+    # identified by Textract in each page.  The result is organized as a list of strings,
+    # each containing the concatenated text of a given page.
+    #
+    def extractTextByPages(self,
+                           textract,
+                           rawPages,
+                           numOfPages):
+
+        # current service limit of Comprehend per page
+        MAX_COMPREHEND_UTF8_PAGE_SIZE = 5000
+        
+        # the textract results json file in S3 has a peculiar structure of
+        # top-level list of block lists, each appear to have a size limit of 1000.
+        # A new one is creatd and appended when the previous blocklist reaches that limit.
+        # Not sure why it is so. A blocklist may contain blocks from different document pages.
+        for blocklist in textract:
+        
+            for block in blocklist['Blocks']:
+        
+                # PAGE block type have no text, so skip those
+                if block['BlockType'] != 'PAGE':
```

Comment:
Correct me if I am wrong, but can't we just get all the data using the LINE type block? 
If yes, we could switch this filter to only process on LINE blocks?

Reply:
there is WORD i saw, but filtering on LINE is probably best then

Reply:
will make the change